test/fixtures/permission-validation/setup.js

```diff
@@ -0,0 +1,484 @@
+  ],
+  InsertOnly: [
+    { Category: 'Category', Access: new Uint64(0b0010) },
+    { Category: 'Person', Subsystem: 'S1', Access: new Uint64(0b0010) },
+    { Category: 'Document', Catalog: 'C1', Access: new Uint64(0b0010) },
+    { Category: 'Catalog', onlyForSubsystem: 'S1', Access: new Uint64(0b0010) },
+    {
+      Category: 'PersonDocument',
+      Subsystem: 'S1',
+      Access: new Uint64(0b0010),
+    },
+    {
+      Category: 'PersonDocument',
+      Catalog: 'C2',
+      Access: new Uint64(0b0010),
+    },
+  ],
+  UpdateOnly: [
+    { Category: 'Category', Access: new Uint64(0b0100) },
+    { Category: 'Catalog', Access: new Uint64(0b0100) },
+    { Category: 'Role', Subsystem: 'S1', Access: new Uint64(0b0100) },
+    { Category: 'SystemUser', Subsystem: 'S1', Access: new Uint64(0b0100) },
+    { Category: 'Person', Subsystem: 'S1', Access: new Uint64(0b0100) },
+    { Category: 'Document', Catalog: 'C1', Access: new Uint64(0b0100) },
+    {
+      Category: 'Permission',
+      onlyForSubsystem: 'S1',
+      Access: new Uint64(0b0100),
+    },
+    {
+      Category: 'PersonDocument',
+      Subsystem: 'S2',
+      Access: new Uint64(0b0100),
+    },
+    {
+      Category: 'PersonDocument',
+      Catalog: 'C2',
+      Access: new Uint64(0b0100),
+    },
+  ],
+  DeleteOnly: [
+    { Category: 'Category', Access: new Uint64(0b1000) },
+    { Category: 'Catalog', Access: new Uint64(0b1000) },
+    { Category: 'Person', Subsystem: 'S1', Access: new Uint64(0b1000) },
+    { Category: 'Document', Catalog: 'C1', Access: new Uint64(0b1000) },
+    {
+      Category: 'PersonDocument',
+      Subsystem: 'S2',
+      Access: new Uint64(0b1000),
+    },
+    {
+      Category: 'PersonDocument',
+      Catalog: 'C1',
+      Access: new Uint64(0b1000),
+    },
+  ],
+  Blocked: [
+    { Category: 'Catalog', Access: new Uint64(0b11111) },
+    { Category: 'Category', Access: new Uint64(0b11111) },
+    { Category: 'Role', Subsystem: 'S1', Access: new Uint64(0b11111) },
+    { Category: 'Role', Subsystem: 'S2', Access: new Uint64(0b11111) },
+    { Category: 'SystemUser', Subsystem: 'S1', Access: new Uint64(0b11111) },
+    { Category: 'SystemUser', Subsystem: 'S1', Access: new Uint64(0b11111) },
+    { Category: 'Person', Subsystem: 'S1', Access: new Uint64(0b11111) },
+    { Category: 'Document', Catalog: 'C1', Access: new Uint64(0b11111) },
+  ],
+};
+
+const rawData = {
+  Document: [
+    { Name: 'documentC1', Catalog: 'C1' },
+    { Name: 'document2C1', Catalog: 'C1' },
+    { Name: 'documentC2', Catalog: 'C2' },
+  ],
+  Person: [
+    { Name: 'personS1', Subsystem: 'S1' },
+    { Name: 'person2S1', Subsystem: 'S1' },
+    { Name: 'personS2', Subsystem: 'S2' },
+  ],
+  PersonDocument: [
+    { Name: 'personDocumentC1S1', Catalog: 'C1', Subsystem: 'S1' },
+    { Name: 'personDocumentC1S2', Catalog: 'C1', Subsystem: 'S2' },
+    { Name: 'personDocumentC2S2', Catalog: 'C2', Subsystem: 'S2' },
+  ],
+};
+
+const flatten = (array, fn) =>
```

Comment:
This function behaves similarly to `Array.prototype.flatMap()`, so I think it should probably be called `flatMap()`.
Also, it may be better to replace it with `common.Iterator.prototype.flatMap()`.